### PR TITLE
Add metric for percent of segments with replicas.

### PR DESCRIPTION
**Pull request by rven1**

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 1]
```diff
@@ -248,6 +250,8 @@ public void runSegmentMetrics() {
           (nReplicasIdealMax > 0) ? (nReplicasExternal * 100 / nReplicasIdealMax) : 100);
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.SEGMENTS_IN_ERROR_STATE,
           nErrors);
+      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_SEGMENTS,
+          (nSegments > 0) ? (100 - (nOffline/nSegments)*100) : 100);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: won't this always result in 100?

I think you need 

(nSegments > 0) ? (100 - (nOffline * 100 /nSegments)) : 100)

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 2]
```diff
@@ -291,6 +295,7 @@ void setStatusToDefault() {
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.NUMBER_OF_REPLICAS, 0);
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_REPLICAS, 0);
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.SEGMENTS_IN_ERROR_STATE, 0);
+      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_SEGMENTS, 0);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: Please add some comments here on what these metrics mean. is this percent of segmens that are Avaialble, or those that are Unavailable? In fact, the PERCENT_OF_REPLICAS should also have been named correctly. PERCENT_REPLICAS_UNAVAILABLE. In this case, it is percent of segments that are unavailable, right?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 3]
```diff
@@ -184,9 +184,11 @@ public void runSegmentMetrics() {
       int nReplicasExternal = -1;
       int nErrors = 0;
       int nOffline = 0;
+      int nSegments = 0;
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: There are too many counters here, and it is not easy to understand the logic anymore. Can you please write a one-liner on what each of these counters represent? thanks

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java (hunk 2 above)
Comment (rven1) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: It is percentage of available replicas.

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/ControllerGauge.java [hunk 4]
```diff
@@ -22,9 +22,13 @@
...
  */
 public enum ControllerGauge implements AbstractMetrics.Gauge {
-  NUMBER_OF_REPLICAS("replicas", false),
-  PERCENT_OF_REPLICAS("percent", false),
+  NUMBER_OF_REPLICAS("replicas", false), // Number of complete replicas of table in external view containing
+                                         // all segments online in ideal state
+  PERCENT_OF_REPLICAS("percent", false), // Percentage of complete online replicas in external view as compared
+                                         // to replicas in ideal state
   SEGMENTS_IN_ERROR_STATE("segments", false),
+  PERCENT_OF_SEGMENTS("segments", false), // Percentage of segments with at least one online replica in external view
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/ControllerGauge.java]: Please rename to PERCENT_SEGMENTS_AVAILABLE

**Pull request merged**